pkg/parse/source_test.go

```diff
@@ -100,6 +100,7 @@ func TestReadConfigFiles(t *testing.T) {
 			}()
 
 			srcState := &sourceState{
+				spec:    nil, // TODO: ???
```

Comment:
What is the TODO?

Reply:
Ideally we'd have some tests that validate the code branches when `sourceState.spec` in not nil. But I haven't dug into those private methods enough to know how they will behave, or even if there's any change behavior and not just a pass through. `TestReadHydratedDirWithRetry` and `TestReadConfigFiles` are affected.

Reply:
Added a more useful TODO comment